### PR TITLE
CBL-4164: Add tests for SELECT * as per spec

### PR DESCRIPTION
**Pull request by borrrden**

### CODE REVIEW

File: src/Couchbase.Lite.Shared/Sync/HTTPLogic.cs [hunk 1]
```diff
@@ -208,7 +208,7 @@ private static string GetUserAgent()
 
 			var runtimePlatform = Service.GetInstance<IRuntimePlatform>();
 			var osDescription = runtimePlatform?.OSDescription ?? RuntimeInformation.OSDescription;
-			var hardware = runtimePlatform != null ? $"; {runtimePlatform.HardwareName}" : "";
+			var hardware = runtimePlatform?.HardwareName != null ? $"; {runtimePlatform.HardwareName}" : "";
```
Comment (bmeike) [inline src/Couchbase.Lite.Shared/Sync/HTTPLogic.cs:211]: total niggle....
would it not be clearer to say:
`var hardware = runtimePlatform?.HardwareName == null ? "" : $"; {runtimePlatform.HardwareName}";`

File: src/Couchbase.Lite.Tests.Shared/DatabaseTest.cs [hunk 2]
```diff
@@ -1530,7 +1530,7 @@ public void TestCookiesExpiration()
             };
 
             foreach (var cookie in noneExpiredCookies) {
-                Db.SaveCookie(cookieStr, uri, false).Should().BeTrue("because otherwise the cookie did not save");
+                Db.SaveCookie(cookie, uri, false).Should().BeTrue("because otherwise the cookie did not save");
```
Comment (bmeike) [inline src/Couchbase.Lite.Tests.Shared/DatabaseTest.cs:1533]: WTF????

File: src/Couchbase.Lite.Tests.Shared/QueryTest.cs [hunk 3]
```diff
@@ -2738,6 +2738,53 @@ public void TestFTSQueryWithJoin()
...
         }
 
+        [Fact]
+        public void TestSelectAllResultKey()
+        {
+            // Section 8.11.10 / 8.12.4 - CBL Helium - Scope and Collection API
+            using var c = Db.CreateCollection("flowers", "test");
+            using var d = Db.GetDefaultCollection();
+
+            var iterations = new List<(string queryID, string resultName)>
```
Comment (bmeike) [inline src/Couchbase.Lite.Tests.Shared/QueryTest.cs:2748]: This really should be called "expectedResultNames" or something halfway intelligible.
 Same with qbIterations

**Review by bmeike — APPROVED** [review]
I'd love to see more descriptive names on those variables.

It all looks right, though.

File: src/Couchbase.Lite.Tests.Shared/DatabaseTest.cs (hunk 2 above)
Comment (borrrden) [reply src/Couchbase.Lite.Tests.Shared/DatabaseTest.cs:1533]: Rebase error 😩

File: src/Couchbase.Lite.Tests.Shared/QueryTest.cs (hunk 3 above)
Comment (borrrden) [reply src/Couchbase.Lite.Tests.Shared/QueryTest.cs]: Updated!

**Pull request merged**